### PR TITLE
internal/appsec: add server.request.method address

### PR DESCRIPTION
**Pull request by eliottness**
### What does this PR do?

This address is now sent to the WAF for further examination.

### Motivation

This address is not used by all the recommended ruleset, but with custom rules we will need to support it.

### Describe how to test/QA your changes

I added a test about a custom rule using this address.

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality.
- [ ] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

File: internal/appsec/waf_test.go [hunk 1]
```diff
@@ -24,6 +24,45 @@ import (
...
+
+	srv := httptest.NewServer(mux)
+	defer srv.Close()
+
+	t.Run("method", func(t *testing.T) {
+		mt := mocktracer.Start()
+		defer mt.Stop()
+
+		req, err := http.NewRequest("POST", srv.URL, nil)
+		if err != nil {
```
Comment (Hellzy) [inline internal/appsec/waf_test.go:50]: Let's go with `require.NoError(...)` for consistency

**Review by Hellzy — COMMENTED** [review]
Looking good 👍

File: internal/appsec/waf_test.go (hunk 1 above)
Comment (eliottness) [reply internal/appsec/waf_test.go]: done 👍

File: internal/appsec/waf.go [hunk 2]
```diff
@@ -210,6 +210,26 @@ func newHTTPWAFEventListener(handle *waf.Handle, addresses []string, timeout tim
 			values := make(map[string]interface{}, len(addresses))
 			for _, addr := range addresses {
 				switch addr {
+				case serverRequestMethodAddr:
```
Comment (Hellzy) [inline internal/appsec/waf.go:213]: This whole chunk shouldn't be added.
You simply need to add the new switch case in the loop at line 160.

**Review by Hellzy — CHANGES_REQUESTED** [review]

File: internal/appsec/waf_test.go [hunk 3]
```diff
@@ -24,6 +24,62 @@ import (
...
+			name:      "method",
+			method:    "POST",
+			ruleMatch: "custom-001",
+		},
+		{
+			name:      "no-method",
+			method:    "GET",
+		},
+	} {
+		t.Run("method", func(t *testing.T) {
```
Comment (Hellzy) [inline internal/appsec/waf_test.go:60]:
```suggestion
		t.Run(tc.name, func(t *testing.T) {
```

**Review by Hellzy — COMMENTED** [review]
Almost there 😅

File: internal/appsec/waf_test.go (hunk 3 above)
Comment (eliottness) [reply internal/appsec/waf_test.go]: done 👍

**Review by Hellzy — APPROVED** [review]
Well done. And congrats on your first product contrib 🎉

**Pull request merged**